sdk_experimental/lib/opentelemetry/sdk/trace/samplers/consistent_probability_tracestate.rb

```diff
@@ -106,7 +106,7 @@ def decimal(str)
           end
 
           def generate_r(trace_id)
-            x = trace_id[8, 8].unpack1('Q>') | 0x3
+            x = trace_id.unpack1('Q>', offset: 8) | 0x3
```

Comment:
Not sure if you are aware but `offset` was added in 3.1, and IIRC otel is supposed to be compatible with Ruby 3.0?

But since the offset is static here, you can use the `@` syntax:

```ruby
>> "abcedfghsdfdsfhdsfjsdkjfdsjfkdlsfjdsfjksdfk".unpack1('Q>', offset: 8)
=> 8314883393651632228
>> "abcedfghsdfdsfhdsfjsdkjfdsjfkdlsfjdsfjksdfk".unpack1('@8Q>')
=> 8314883393651632228
```

`offset` was added to avoid allocating a new pattern string every time when the offset is dynamic.